src/org/labkey/test/components/glassLibrary/components/MultiMenu.java

```diff
@@ -60,16 +62,67 @@ public void doMenuAction(List<String> pathToAction)
         }
     }
 
+    private List<String> waitForDataThenGetMenuText()
+    {
+        List<WebElement> topMenuList;
+        List<String> menuText = new ArrayList<>();
+
+        boolean stale;
+        boolean loading;
+        int tries = 1;
+
+        do {
+
+            stale = false;
+            loading = false;
+
+            try
+            {
+
+                topMenuList = Locator.tagWithAttribute("a", "role", "menuitem").findElements(this);
+                menuText = new ArrayList<>();
+
+                for (WebElement menuItem : topMenuList) {
+
+                    String menuItemText = menuItem.getText();
+
+                    // Check to see if we are still loading data.
+                    if(menuItemText.toLowerCase().contains("loading"))
+                        loading = true;
+
+                    menuText.add(menuItem.getText());
+
+                }
+
+            }
+            catch(StaleElementReferenceException staleExc)
+            {
+                // This happens in the time between the change of the menu content from containing "loading"
+                // to having data (like "sample sets").
+                stale = true;
+            }
```

Comment:
Make sure that it wasn't the menu itself that went stale so this doesn't keep looping futilely.
```
catch (..)
{
    [..]
    getComponentElement().isDisplayed(); // will throw an uncaught 'StaleReferenceException' if the entire menu went stale
    stale = true;
}
```

Reply:
Updated.